pkg/metrics/metrics_test.go

```diff
@@ -64,6 +67,31 @@ func TestRegisterCustomMetrics(t *testing.T) {
 	}
 }
 
+func TestMetricsHandler(t *testing.T) {
+	// Create a request to pass to our handler. We don't have any query parameters, so we'll
+	// pass 'nil' as the third parameter.
+	req, err := http.NewRequest(http.MethodGet, "/metrics", nil)
+	require.NoError(t, err)
+
+	// Create handler instance.
+	RegisterCustomMetrics()
+
+	t.Run("valid metrics json", func(t *testing.T) {
+		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+		e := echo.New()
+		rec := httptest.NewRecorder()
+		ctx := e.NewContext(req, rec)
+
+		//when
+		err := prometheusHandler(ctx)
+		require.NoError(t, err)
+		// then
+		// check the status code is what we expect, and the content-type
+		require.Equal(t, http.StatusOK, rec.Code)
+		require.Equal(t, "text/plain; version=0.0.4; charset=utf-8", rec.Header().Get("Content-Type"))
```

Comment:
does it make sense to verify that the body also contains the actual registered metric? you know, just check that the name of the metric is there, no need to verify the actual values...

Reply:
removed